gitlab.go

```diff
@@ -505,8 +516,20 @@ func newClient(httpClient *http.Client) *Client {
 		RetryMax:     30,
 	}
 
-	// Set the default base URL.
-	_ = c.SetBaseURL(defaultBaseURL)
+	// Set the default base URL and apply any given client options.
+	for _, fn := range append([]ClientOptionFunc{WithBaseURL(defaultBaseURL)}, options...) {
+		if fn == nil {
+			continue
+		}
+		if err := fn(c); err != nil {
+			return nil, err
+		}
+	}
+
+	// Configure the rate limiter.
+	if err := c.configureLimiter(); err != nil {
```

Comment:
Perhaps make it an option too?

Also, I think longer term it'd be better to make the request to get the limits on first client use. I'd say it's quite unexpected to have I/O happen as part of client construction. [`sync.Once`](https://golang.org/pkg/sync/#Once) is a perfect fit for the job. Or even better - configure with no limiter initially and then after the first successful request initialize the limiter properly. On each subsequent request check that the limiter is initialized to the correct values - they may change during the lifetime of a client.

Reply:
Another thing to consider - currently an I/O error to get the limit values will prevent the client from being constructed and this is not great. The remote side may be unavailable but the client for it should be constructable. In my  experience applications are typically prepared to handle I/O errors when using clients but treat client construction errors as fatal.

I once was dealing with a DB client library that would throw an exception when a client was constructed while the DB was unavailable and it was a huge pain point. I had to build a wrapper around the library to handle this. On each wrapper client use it'd check if the real client was already constructed and construct one if not. Was a PITA.

Reply:
I partly agree... Especially on the point that creating the client should not fail with an I/O error, so I changed the behavior a little to make sure that will not happen.

I initially intended `configureLimiter()` to be able to fail silently. So the error can actually be ignored as then the limiter will be configured as unlimited (i.e. disabled). That got lost in the changes somehow, but it's now used as such again.

I recognize that this could maybe be improved in the future, but let's keep it like this for now and first get some feedback for a few months (the limiter logic is super new).